abstract_domains/numerical/octagon.py

```diff
@@ -0,0 +1,578 @@
+from copy import deepcopy
+from enum import Enum
+from functools import reduce
+
+from abstract_domains.lattice import BottomMixin
+from abstract_domains.numerical.dbm import IntegerCDBM
+from abstract_domains.numerical.interval import IntervalLattice, IntervalDomain
+from abstract_domains.numerical.numerical import NumericalMixin
+from abstract_domains.state import State
+from core.expressions import *
+from typing import List, Set, Tuple, Union
+from math import inf, isinf
+
+from abstract_domains.numerical.linear_forms import SingleVarLinearForm, LinearForm, InvalidFormError
+
+from core.expressions_tools import ExpressionVisitor, ExpressionTransformer, \
+    make_condition_not_free, simplify
+
+Sign = UnaryArithmeticOperation.Operator
+PLUS = Sign.Add
+MINUS = Sign.Sub
```

Comment:
Why are you reusing the operators of a unary arithmetic operation in this context? It also seems a bit hacky to have this defined at the top level like this.